File: Hanoi/PHP/Hanoi.php
<?php 
$time_start = microtime(true); 

function hanoi($topN, $from, $to, $inter){
    if($topN != 0){
        hanoi($topN -1, $from, $inter, $to);
        //echo "Moving Disk  $topN from $from to $to \n";
        hanoi($topN -1, $inter, $to, $from);
    }
}

$nDisks = $argv[1];
hanoi($nDisks, "Depart(0)", "Arrivee(2)", "Milieu(1)");
      

$time_end = microtime(true);
(int)$execution_time = (int)(($time_end - $time_start) * 1000);
echo "execution time : ${execution_time}ms \n"
?>

File: Hanoi/Scala/hanoi.scala
object hanoi {
  
  def hanoi(topN:Int,from:String,to:String,inter:String){
    if(topN != 0){
      hanoi(topN-1,from,inter,to);
      //System.out.println("Moving Disk " + topN + " from " + from + " to " + to);
      hanoi(topN-1,inter,to,from);
    }
  } 
  
  def main(args: Array[String]) {
      val nDisks = args(0).toInt;
      val now = System.currentTimeMillis();
      hanoi(nDisks, "Depart(0)", "Arrivee(2)", "Milieu(1)");
      val end = (System.currentTimeMillis() - now);
      System.out.println("execution time : "+end+" ms ");
  }
}
